### ISSUE TITLE
Feature request: allow passing a --config option to `cog` to use a different cog.yaml

### ISSUE THREAD

**Issue by ericallam** [replicate/cog#257]
It would be nice to be able to have multiple cog.yaml files in a repo and be able to pass a `--config` option to the cog cli like so:

```bash
$ cog build -t image_name --config cog.dev.yaml
```

**Comment by bfirsh** [comment]
👍 Great idea. Thanks for the suggestion!

Perhaps the option could be `--file/-f` for consistency with `docker build`.

For a long time we resisted adding the option to select a file to Docker. There was a reason for it, but it is fuzzy in my head. Let me try and recall why that was... @aanand do you remember?

**Comment by aanand** [comment]
I don't think we resisted the `--file` argument to `docker-compose` necessarily, but there was certainly a lot of hesitancy over allowing _multiple_ files to be passed in and the semantics thereof. If that's a moot point here, I don't think you have much to worry about.

**Comment by aanand** [comment]
I think that's right. It probably never mattered whether `docker build` with no arguments was a reliably repeatable command, because in practice it's almost always invoked by some CI system, build script, `docker-compose` file, etc, rather than by hand.

The question, then, is whether `cog` is more of a `docker build`-style command (overwhelmingly invoked by computers) or a `docker-compose`-style command (frequently invoked by humans)?

**Comment by aanand** [comment]
(that said, it might not matter - as you point out, adding a `--file` argument to `docker-compose` doesn't seem to have resulted in a mass of people naming their Compose files something else.)

**Comment by bfirsh** [comment]
`cog` is a human thing like Compose, so I guess this will be fine. Thanks for your thoughts!

**Comment by hardikdava** [comment]
Is it supported now?

**Comment by markphelps** [comment]
This feature request has been inactive for over a year. While the team previously agreed that adding a `--file` / `-f` option is a good idea, it is not yet supported and there is no active work planned. I'm closing this as stale to keep the backlog manageable. If you're still interested, we'd welcome a PR implementing it.
